apps/gdalalg_raster_pipeline.h

```diff
@@ -18,13 +18,50 @@
 
 //! @cond Doxygen_Suppress
 
+/************************************************************************/
+/*                  GDALRasterPipelineStepRunContext                    */
+/************************************************************************/
+
+class GDALRasterPipelineStepAlgorithm;
+
+class GDALRasterPipelineStepRunContext
+{
+  public:
+    GDALRasterPipelineStepRunContext() = default;
+
+    GDALProgressFunc m_pfnProgress = nullptr;
+    void *m_pProgressData = nullptr;
+    GDALRasterPipelineStepAlgorithm *m_poNextStep = nullptr;
+
+  private:
+    CPL_DISALLOW_COPY_ASSIGN(GDALRasterPipelineStepRunContext)
+};
+
 /************************************************************************/
 /*                GDALRasterPipelineStepAlgorithm                       */
 /************************************************************************/
 
 class GDALRasterPipelineStepAlgorithm /* non final */ : public GDALAlgorithm
 {
+  public:
+    const GDALArgDatasetValue &GetOutputDataset() const
+    {
+        return m_outputDataset;
+    }
+
+    const std::string &GetFormat() const
```

Comment:
```suggestion
    const std::string &GetOutputFormat() const
```
just to clearly distinguish it from the input format.